### DISCUSSION TITLE
retrieve a socket connection directly using its socketID with O(1) time complexity

### DISCUSSION THREAD

**Discussion by jamie0xgitc0decat** [socketio/socket.io#4918] [Q&A]
I am currently exploring the capabilities of Socket.IO and have a question regarding the efficiency of socket retrieval. Specifically, is it possible to retrieve a socket connection directly using its socketID with O(1) time complexity?

I am aware of the existing io.fetchSockets() method, which retrieves a list of all connected sockets. However, this approach seems less efficient for scenarios where direct access to a specific socket is required.

Any insights or guidance on whether Socket.IO supports a more direct and efficient method for socket retrieval would be greatly appreciated.

Thank you for your assistance.

**Comment by darrachequesne** [comment]
Hi! Yes, you can use the following method:

```js
const socket = io.of("/").sockets.get(theSocketId);
```

This won't work within a cluster though, as the `sockets` Map only contains the connections that are currently on this server instance.
